### DISCUSSION TITLE
core venv vs docker, quanta ram occupano?!

### DISCUSSION THREAD

**Discussion by 50494554524F** [indomus/forum#1473] [HUB: Home Assistant]
sto provando core in previsione di poterla usare su raspberry zero 2 w. 
sono riuscito a farla funzionare in macchina virtuale con debian minimal, 512mb ram, core, samba e ssh. 
dovrebbe esserci tutto quello che serve, occupa intorno ai 350mb e, quindi, entrarci anche sul raspberry.
prima di studiare anche docker, sapreste dirmi quando verrebbe ad occupare e se c'è troppo rallentamento rispetto a venv?!
grazie

**Comment by snakuzzo** [comment]
Le prestazioni non vanno a degradare con Docker...anzi...quando ho fatto i confronti tempo fa ho notato un'occupazione minore di risorse.
HA sostanzialmente gira pure su uno smartphone Android...ovvio che poi dipende da quello che cominci a metter su. Se ad esempio cominci a mettere "roba" che chiede un uso spropositato di CPU (vedi telecamere con conseguente transcodifica), ovvio che comincia a diventare pesante e sicuramente hai bisogno di HW migliore e/o dedicato.
Tra le due soluzioni ti consiglio a mani basse Docker per tutta una serie di motivi:
L'immagine che andrai a scaricare e che userai per la creazione del container è autoconsistente. Non hai bisogno di star appresso ad eventuali aggiornamenti di librerie di sistema (tranne rarissimi casi), Python, etc (e credimi che questo capita ogni volta che viene deprecata una versione di Python in favore di quella più recente)
Poi avrai una maggiore semplicità d'uso. Considera che con Docker + docker-compose ho migrato nel giro di mezz'ora da un server all'altro senza alcuna "pippa mentale" (ovviamente devi conoscere un minimo Docker...più che altro i concetti sui volumi e network)

**Reply by Scialla** [reply]
Concordo, anche se devo dire che con docker, se fai uso di custom component, ogni volta che aggiorni HA se li deve riscaricare, mentre in venv no

**Reply by snakuzzo** [reply]
Sii più preciso...Non è che "riscarica il custom component".
Quello che intendi tu non riguarda solo i custom, ma qualsiasi integrazione che necessiti di installazione di pacchetti python non inclusi di default nell'immagine Docker che scarichi da github container registry o da dockerhub

EDIT: Però permettimi di dire che è un particolare che si può pure tralasciare in confronto ai vantaggi che noterai

**Reply by Scialla** [reply]
ripeto che sono concorde, però il riavvio dopo l'aggiornamento (su rpi3b) passa da 75s di media a circa 499s :)

chissene eh, :)

**Reply by 50494554524F** [reply]
allora provato docker, la ram occupata è più o meno la stessa... dovrei, sempre che i prezzi riscendano, solo catturare i termometri in garage; sono bluetooth ed ho un po' di difficoltà a trovare la posizione senza dover rispostare tutto. una volta funzionante penso resterà così, non è fondamentale aggiornarlo.
userò docker visto che è più veloce da installare/aggiornare... mi toccherà imparare anche docker 😄 


OT: riguardo core/docker... dove erano documentate /config e /local che non lo ritrovo?!

**Reply by snakuzzo** [reply]
È sempre un tempo relativo a quanta schifezza aggiungi su HA...e tu sei pieno di schifezza ;)

**Reply by snakuzzo** [reply]
> 
> OT: riguardo core/docker... dove erano documentate /config e /local che non lo ritrovo?!

Che intendi?

**Reply by Scialla** [reply]
> È sempre un tempo relativo a quanta schifezza aggiungi su HA...e tu sei pieno di schifezza ;)

vero! :)

Per quanto riguarda il BT puoi sempre usare un ESP32 se ti arriva la wifi

**Reply by 50494554524F** [reply]
> > OT: riguardo core/docker... dove erano documentate /config e /local che non lo ritrovo?!
> 
> Che intendi?

ieri ho perso una sacco di tempo con i percorsi degli include e delle risorse... 

/config punta alla posizione in cui si trovano i file di configurazione, /home/nomeutente/homeassistant-core nel mio caso
/local punta a /home/nomeutente/homeassistant-core/www dove sono i componenti di lovelace

poi ho usato questi perché li uso su hassos ma non ritrovo dove son documentati...

**Reply by 50494554524F** [reply]
> Per quanto riguarda il BT puoi sempre usare un ESP32 se ti arriva la wifi

lo sviluppatore che si occupa del firmware dei termometri ha detto che esp si perde "tutti" i pacchetti che dovrebbe ricevere dai termometri... sono gli xiaomi thermometer 2

**Reply by snakuzzo** [reply]
> /config punta alla posizione in cui si trovano i file di configurazione, /home/nomeutente/homeassistant-core nel mio caso
> /local punta a /home/nomeutente/homeassistant-core/www dove sono i componenti di lovelace

in realtà ti basta sapere questo.
Devi far conto che nelle config di HA dovrai usare sempre i path

```
/config
/local
```

Se entri invece nel container Docker (`docker exec -it homeassistant bash`) vedrai che hai la directory `/config` e quindi anche `/config/www`.

Nelle configurazioni di HA devi ragionare sempre come se stessi all'interno di questo container e non sull'host

**Reply by 50494554524F** [reply]
tanto per passare tempo ho provato ad installare hassos con 512mb di ram.
con grande sorpresa, parte, si posso installare subito gli addon samba e ssh, funziona bene.
naturalmente, appena si aggiunge altro di troppo si impalla.
ma... aggiungendo un file di swap più grande torna a funzionare splendidamente.
devo provare ad installare tutto quello che ho sul raspberry e vedere se tiene, ma per un paio di termometri andrà bene sicuramente... ancora meno lavoro da fare, posso studiare docker con calma
